util/imageutil/config.go

```diff
@@ -211,16 +212,7 @@ func childrenConfigHandler(provider content.Provider, platform platforms.MatchCo
 			if err := json.Unmarshal(p, &index); err != nil {
 				return nil, err
 			}
-
-			if platform != nil {
-				for _, d := range index.Manifests {
-					if d.Platform == nil || platform.Match(*d.Platform) {
-						descs = append(descs, d)
-					}
-				}
-			} else {
-				descs = append(descs, index.Manifests...)
-			}
+			descs = append(descs, index.Manifests...)
```

Comment:
`platform.Match` condition should still be here. `LimitManifests` does not call `Match` and if we skip this it could lead to wrong resolution (eg. for nested indexes).

Reply:
@cpuguy83 @tonistiigi I've pushed an update to add this block back in to keep the original condition.

Reply:
Sorry I totally missed this comment.